src/components/asset-list/RecyclerAssetList2/FastComponents/FastBalanceCoinRow.tsx

```diff
@@ -0,0 +1,330 @@
+import React, { useCallback, useRef } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { CoinIconIndicator } from '../../../../components/coin-icon';
+import { Icon } from '../../../../components/icons';
+import { ButtonPressAnimation } from '../../../animations';
+
+import { initialChartExpandedStateSheetHeight } from '../../../expanded-state/asset/ChartExpandedState';
+import { ExtendedState } from '../core/RawRecyclerList';
+import FastCoinIcon from './FastCoinIcon';
+import { Text } from '@rainbow-me/design-system';
+import {
+  useAccountAsset,
+  useCoinListFinishEditingOptions,
+} from '@rainbow-me/hooks';
+import Routes from '@rainbow-me/routes';
+import { borders, colors, padding, shadow } from '@rainbow-me/styles';
+
+interface CoinCheckButtonProps {
+  isHidden: boolean;
+  isPinned: boolean;
+  onPress: () => void;
+  uniqueId: string;
+  theme: any;
+}
+
+const CoinCheckButton = React.memo(function CoinCheckButton({
+  isHidden,
+  isPinned,
+  onPress,
+  uniqueId,
+  theme,
+}: CoinCheckButtonProps) {
+  const { selectedItems } = useCoinListFinishEditingOptions();
```

Comment:
It's used only inside the checkbox component, so even if it changes - we rerender only that component. otherwise selectedItems will cause rerender for the whole FastBalanceCoinRow.
we make sure that firing toggleSelectedCoin will only rerender checkboxes.